plugins/patriot-hadoop/lib/patriot_hadoop/ext/hive.rb

```diff
@@ -16,19 +16,27 @@ class HiveException < Exception; end
 
 
       def execute_hivequery(hql_file, output_file=nil, user=nil)
-        command = "hive -f \"#{hql_file}\""
         unless user.nil?
           if user !~ /^[a-z_][a-z0-9_]{0,30}$/
             raise HiveException, "Invalid username" 
           end
-          command = "sudo -u #{user} #{command}"
+          command = "sudo -u #{user} hive -f \"#{hql_file}\" #{_compress_option(File.extname(output_file))}"
```

Comment:
Does it work in case user == nil ?

Reply:
Fixed and checked with the following pbc.

```
hive {
  name_suffix 'simple_hive_job_#{_date_}'
  output_prefix '/tmp/execute_simple_hive_job_#{_date_}'
  compression 'gzip'
  hive_ql <<-EOS
    SELECT * FROM some_table WHERE dt='#{_date_}' LIMIT 3
  EOS
}
```